pyrene/src/components/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,321 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+
+import ReactDatePickerWrapper from './ReactDatePickerWrapper/ReactDatePickerWrapper';
+import DateTimeInput from './DateTimeInput/DateTimeInput';
+
+import {
+  getFutureDate, customDateFormat, convertToUTCtime,
+  convertToZoneTime, customStringToDate, getErrors,
+  hasDateError, hasTimeError, getClientTimeZone,
+} from '../../utils/DateUtils';
+
+type OnFunction = (value?: number) => void;
+
+export interface DateTimePickerProps{
+  /**
+   * Calendar is opened on input component render
+   */
+  calendarOpened?: boolean,
+  /**
+   * Close calendar on date select
+   */
+  closeOnSelect?: boolean,
+  /**
+   * Date format used by component
+   */
+  dateFormat?: string,
+  /**
+   * Boolean to toggle time display
+   */
+  dateOnly?: boolean,
+  /**
+   * Date object that represents the end date of the component
+   */
+  endDate?: Date,
+  /**
+   * Display the component inline
+   */
+  inline?: boolean,
+  /**
+   * String label of the component
+   */
+  label?: string,
+  /**
+   * Unix timestamp that represents the maximum date allowed by the component
+   */
+  maxDateTime?: number,
+  /**
+   * Unix timestamp that represents the minimum date allowed by the component
+   */
+  minDateTime?: number,
+  /**
+   * Name that can be used to uniquely identify the component
+   */
+  name?: string,
+  /**
+   * Function to handle onBlur event
+   */
+  onBlur?: () => OnFunction,
+  /**
+   * Calendar open callback
+   */
+  onCalendarOpen?: () => void,
+  /**
+   * Click outside componet callback
+   */
+  onClickOutside?(event: React.MouseEvent<HTMLDivElement>): void,
+  /**
+  * Function to handle onChange event
+  */
+  onChange: OnFunction,
+  /**
+   * Input selects the end date of a range
+   */
+  selectEnd?: boolean,
+  /**
+  * Input selects the start date of a range
+  */
+  selectStart?: boolean,
+  /**
+  * Date object that represents the start date of the component
+  */
+  startDate?: Date,
+  /**
+   * Date format used by component
+   */
+  timeFormat?: string,
+  /**
+   * Unix timestamp to initialize component with
+   */
+  timeStamp?: number | null
+  /**
+   * This is must be a IANA time zone string
+   */
+  timeZone?: string,
+}
+
+/**
+ * A component for selecting date and time.
+ */
+const DateTimePicker: React.FC<DateTimePickerProps> = ({
+  calendarOpened,
+  closeOnSelect = true,
+  dateFormat = 'dd.MM.yyyy',
+  dateOnly = false,
+  endDate,
+  inline = false,
+  label,
+  maxDateTime = getFutureDate({ years: 1 }),
+  minDateTime = 0,
+  name,
+  onCalendarOpen,
+  onChange,
+  onClickOutside,
+  selectEnd,
+  selectStart,
+  startDate,
+  timeFormat = ' HH:mm',
+  timeStamp,
+  timeZone,
```

Comment:
What I mean is the following : 
```
  ...
  timeFormat = ' HH:mm',
  timeStamp,
  timeZone = getClientTimeZone(),
```
So you do not need a dedicated **state** for the `timeZone`.

Reply:
✅